### PR TITLE
PROMO-1051: [add] Promotion Settings v3 REST contract

### PR DESCRIPTION
**Pull request by jpajar**
# [PROMO-1051]

## What changed?

* Added documentation for future Promotion Settings REST API

## Release notes draft
* Added documentation for future Promotion Settings REST API

[PROMO-1051]: https://bigcommercecloud.atlassian.net/browse/PROMO-1051?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by BC-SEven** [comment]
Should we add to the description for the fields that can only be edited for enterprise accounts? Also might be nice to document a `403` error response for when a client tries to change settings on a store that isn't enterprise.

File: reference/promotion_settings.v3.yml
Comment (matt-evangelidis) [inline reference/promotion_settings.v3.yml:107]: Given we have a few examples of fields specific to enterprise accounts, it's probably worthwhile defining what the behaviour is for non-enterprise accounts somewhere in this schema. I believe we brought this up when fleshing out some related tickets in a refinement session.

Some options: 
- Do we throw an error if a non-enterprise merchant tries to pass a non-default setting to an enterprise-only field?
- Do we not persist the change and just ignore it?
- Do we ignore it but offer some kind of feedback in the payload about them attempting to edit an enterprise-only field?

Comment (jpajar) [reply reference/promotion_settings.v3.yml:107]: @bc-vincent-zhao did we happen to officially decide on an approach?

Comment (BC-SEven) [reply reference/promotion_settings.v3.yml:107]: From memory we decided to throw an error if the value in the PUT request is different to the current value for non enterprise plans.

File: reference/promotion_settings.v3.yml
Comment (jpajar) [inline reference/promotion_settings.v3.yml:88]: Note for Docs team: Right now for Orders / Promotions documentation (as seen in screenshot) we aren't providing relevant error examples. What's the recommended work around for that?
image

File: reference/promotion_settings.v3.yml
Comment (bc-vincent-zhao) [reply reference/promotion_settings.v3.yml:107]: what @BC-SEven said ^ seems sensible.

File: reference/promotion_settings.v3.yml
Comment (bc-traciporter) [inline reference/promotion_settings.v3.yml:65]:
```suggestion
            example:
```

File: reference/promotion_settings.v3.yml
Comment (bc-traciporter) [reply reference/promotion_settings.v3.yml:88]: Hi @jpajar we can add them. Is this your request?

File: reference/promotion_settings.v3.yml
Comment (bc-traciporter) [inline reference/promotion_settings.v3.yml:107]:
```suggestion
            - Note: This field can only be edited for Enterprise Accounts.
```

File: reference/promotion_settings.v3.yml
Comment (bc-traciporter) [inline reference/promotion_settings.v3.yml:115]:
```suggestion
            - Note: This field can only be edited for Enterprise Accounts.
```

File: reference/promotion_settings.v3.yml
Comment (bc-traciporter) [inline reference/promotion_settings.v3.yml:122]:
```suggestion
            - Note: This field can only be edited for Enterprise Accounts.
```

**Review by bc-traciporter — COMMENTED** [review]
I left a few comments.

File: reference/promotion_settings.v3.yml
Comment (jpajar) [reply reference/promotion_settings.v3.yml:88]: Hi @bc-traciporter that would be much appreciated, thanks!

File: reference/promotion_settings.v3.yml
Comment (donald-nguyen-bc) [inline reference/promotion_settings.v3.yml:42]: probably we don't need allOf here ?

File: reference/promotion_settings.v3.yml
Comment (donald-nguyen-bc) [inline reference/promotion_settings.v3.yml:49]: "Authentication Required" probably should be 401

```suggestion
          description: Bad request.
```

File: reference/promotion_settings.v3.yml
Comment (donald-nguyen-bc) [inline reference/promotion_settings.v3.yml:56]: probably we should mention about those enterprise-only settings in this description so it would be easier to notice

File: reference/promotion_settings.v3.yml
Comment (donald-nguyen-bc) [inline reference/promotion_settings.v3.yml:79]: probably can remove this

File: reference/promotion_settings.v3.yml
Comment (donald-nguyen-bc) [inline reference/promotion_settings.v3.yml:128]: not in scope of this PR, just a question to team: if a store downgrades from enterprise plan to pro plan, should we reset this value to 1 automatically ?

Comment (matt-evangelidis) [reply reference/promotion_settings.v3.yml:128]: I'd imagine we already do this in the current implementation? Probably a question for Just Way if he knows of any existing cases.

**Comment by donald-nguyen-bc** [comment]
Overall LGTM, just some minor comments

File: archive/storefront/blueprint/themes/global-variables.mdx
Comment (matt-evangelidis) [inline archive/storefront/blueprint/themes/global-variables.mdx:11]: Was this your change, @jpajar? Seems a little out of scope for our purposes here 😅

**Review by matt-evangelidis — APPROVED** [review]
Looks good to me, @jpajar. Nicely done 👏👏👏

File: archive/storefront/blueprint/themes/global-variables.mdx
Comment (jpajar) [reply archive/storefront/blueprint/themes/global-variables.mdx:11]: Oh that's weird... it's not my changes but something must have happened with the rebase.

File: reference/promotion_settings.v3.yml
Comment (bc-jwk) [reply reference/promotion_settings.v3.yml:129]: Yes! We definitely should.

Comment (donald-nguyen-bc) [reply reference/promotion_settings.v3.yml:129]: cool, do we have a ticket for it already ?

File: reference/promotion_settings.v3.yml
Comment (jpajar) [reply reference/promotion_settings.v3.yml:94]: Thanks @bc-traciporter , I will add comments and more information on that ticket. In the mean time if you have any further feedback re: this open PR please lmk!

Comment (jpajar) [reply reference/promotion_settings.v3.yml:94]: Hey @bc-traciporter just wanted to ask if this PR needs an approval from the dev-docs team?

**Comment by bc-vincent-zhao** [comment]
@bc-traciporter thanks for reviewing 🙇 Once the PR is approved by your team, what would be the next step ? we're still working on the actual implementation for the API at the moment, so it shouldn't be published yet. Given it's entirely a new file that doesn't have reference to others, would it be safe to merge as is ? or is it preferable to defer the merge until the feature is fully done ? Thanks

**Comment by bc-traciporter** [comment]
> @bc-traciporter thanks for reviewing 🙇 Once the PR is approved by your team, what would be the next step ? we're still working on the actual implementation for the API at the moment, so it shouldn't be published yet. Given it's entirely a new file that doesn't have reference to others, would it be safe to merge as is ? or is it preferable to defer the merge until the feature is fully done ? Thanks

@bc-vincent-zhao I added the "do not merge" label to this PR so we know not to merge this PR. Since this is a new article please see the [google doc](https://docs.google.com/document/d/1anFDKMK-hwvV4UvQFVc9h1fdRuWHleeVvFnbvGiKzfc/edit?usp=sharing) for adding new articles and menus to the dev center.

File: reference/promotion_settings.v3.yml
Comment (bc-traciporter) [reply reference/promotion_settings.v3.yml:94]: Yes I will approve the PR.

**Review by bc-traciporter — DISMISSED** [review]
Approved

File: reference/promotion_settings.v3.yml
Comment (bc-vincent-zhao) [inline reference/promotion_settings.v3.yml:135]: @jpajar would you please update this to 15 as per https://github.com/bigcommerce/bigcommerce/pull/58981#discussion_r1687266855 ? thanks.

**Review by bc-traciporter — COMMENTED** [review]
I made a few comments.

**Review by bc-traciporter — APPROVED** [review]
Approved

**Pull request merged**
